native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Password.java

```diff
@@ -135,4 +149,318 @@ public static Object generateSalt(long workFactor) {
+                                 StringUtils.fromString(PasswordUtils.DEFAULT_PBKDF2_ALGORITHM));
+    }
+    
+    /**
+     * Generate a salt string for PBKDF2 with custom parameters.
+     *
+     * @param iterations number of iterations
+     * @param algorithm HMAC algorithm to use (SHA1, SHA256, SHA512)
+     * @return formatted salt string or error
+     */
+    public static Object generateSaltPBKDF2(long iterations, BString algorithm) {
+        try {
+            String alg = algorithm.getValue();
+            
+            Object validationError = PasswordUtils.validatePBKDF2Iterations(iterations);
+            if (validationError != null) {
+                return validationError;
+            }
+            
+            validationError = PasswordUtils.validatePBKDF2Algorithm(alg);
+            if (validationError != null) {
+                return validationError;
+            }
+            
+            byte[] salt = PasswordUtils.generateRandomSalt();
+            String saltBase64 = Base64.toBase64String(salt);
+            
+            String result = PasswordUtils.formatPBKDF2Salt(alg, iterations, saltBase64);
+            
+            return StringUtils.fromString(result);
+        } catch (Exception e) {
+            return CryptoUtils.createError("Error occurred while generating PBKDF2 salt: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Generate PBKDF2 hash using the given parameters.
+     *
+     * @param password password to hash
+     * @param salt salt to use
+     * @param iterations number of iterations
+     * @param algorithm HMAC algorithm to use (SHA1, SHA256, SHA512)
+     * @return hash byte array
+     * @throws NoSuchAlgorithmException if algorithm is not available
+     * @throws InvalidKeySpecException if key specification is invalid
+     */
+    private static byte[] generatePBKDF2Hash(String password, byte[] salt, int iterations, String algorithm)
+            throws NoSuchAlgorithmException, InvalidKeySpecException {
+        
+        String hmacAlgorithm = algorithm.toUpperCase(Locale.ROOT);
+        String pbkdf2Algorithm;
+        int keyLength;
+        
+        switch (hmacAlgorithm) {
+            case "SHA1":
+                pbkdf2Algorithm = "PBKDF2WithHmacSHA1";
+                keyLength = 20; // SHA-1 produces 160-bit (20-byte) hash
+                break;
+            case "SHA256":
+                pbkdf2Algorithm = "PBKDF2WithHmacSHA256";
+                keyLength = 32; // SHA-256 produces 256-bit (32-byte) hash
+                break;
+            case "SHA512":
+                pbkdf2Algorithm = "PBKDF2WithHmacSHA512";
+                keyLength = 64; // SHA-512 produces 512-bit (64-byte) hash
+                break;
+            default:
+                throw new NoSuchAlgorithmException("Unsupported algorithm: " + algorithm);
+        }
+        
+        PBEKeySpec spec = new PBEKeySpec(password.toCharArray(), salt, iterations, keyLength * 8);
```

Comment:
Why do we need to multiply keyLength by 8? It seems like a magic number? Shall we add a comment or introduce a Variable/constant?